### PR TITLE
Add `garden search` command

### LINKED ISSUE #16: Find Garden by Metadata
_As a data consumer I want to be able to find Gardens based on their metadata so I can identify models that meet my needs_

# Description
```shell
% garden-ai garden search --author=mendal
```

Same command line args as garden create.

Results in pretty-printed top-level garden metadata

# Assumptions:
1. A new `search` method will be added to the GardenClient
2. The search will be initiated from the GardenClient and not require an intermediate server
3. Search doesn't require membership in any particular GlobusGroup
4. Determine what metadata will make this an interesting demo and add to the model
5. Results will be dumped to console as JSON
6. Query will accept the GlobusSearch query string

# Acceptance Criteria
Given. I have published a Garden When I submit a query Then I see the JSON

### PR DESCRIPTION
**Pull request by WillEngler**
Closes #16

## Overview

Adds a `garden search` command. For now it just prints the first page of responses (so 10 responses) to the console.

## Discussion

I've started with the boolean string style of syntax that Globus Search supports. So we're passing Globus Search clauses connected by "AND"s concatenated into a big string. Users can also pass their own search string with `--raw-query` if they feel so inclined. 

There is a more advanced method where you can construct a JSON document that describes your query. That lets you perform faceted search and other more advanced features you can't use with the string-only approach. Overall I think searching will mostly be done on the web frontend so I feel ok not making a very feature-rich CLI search experience.

## Testing

Tested this against the test index by issuing the new command. Also added new CLI tests.

----
:books: Documentation preview :books:: https://garden-ai--116.org.readthedocs.build/en/116/

### CODE REVIEW

File: garden_ai/app/garden.py [hunk 1]
```diff
@@ -59,19 +33,6 @@ def garden():
...
-        dir_okay=True,
-        file_okay=False,
-        writable=True,
-        readable=True,
-        resolve_path=True,
-        help=(
-            "(Optional) if specified, this generates a directory with subfolders to help organize the new Garden. "
-            "This is likely to be useful if you want to track your Garden/Pipeline development with GitHub."
-        ),
-    ),
```
Comment (WillEngler) [inline garden_ai/app/garden.py:74]: @OwenPriceSkelly You mentioned in the summer student meeting that you were open to having students get rid of this ... so I took that as a signal that it was ok to get rid of this while I was in here.

Comment (OwenPriceSkelly) [reply garden_ai/app/garden.py:74]: oh yeah more than ok getting rid of this part now

File: garden_ai/app/garden.py [hunk 2]
```diff
@@ -182,6 +143,74 @@ def create(
...
+    ),
+    raw_query: Optional[str] = typer.Option(
+        None,
+        help=(
+            "Form your own Globus Search query directly. It will be passed to Search in advanced mode."
+            "See https://docs.globus.org/api/search/reference/get_query for more details."
+        ),
+    ),
+):
+    """Queries the Garden search index and prints matching results. All query components are ANDed together.
```
Comment (OwenPriceSkelly) [inline garden_ai/app/garden.py:185]: Really like this help text, works as CLI help but even as a less structured docstring it clarifies what the actual code is doing too

File: garden_ai/app/garden.py [hunk 3]
```diff
@@ -182,6 +143,74 @@ def create(
...
+    So if you say `garden-ai garden search --description "foo" --title "bar"` you will get results
+    for gardens that have "foo" in their description and "bar" in their title.
+    """
+    client = GardenClient()
+    if raw_query:
```
Comment (OwenPriceSkelly) [inline garden_ai/app/garden.py:190]: Should we mention that the raw_query option is mutually exclusive with other options, or is that pretty obvious? I can't really picture the user who wants to pass the raw query string and simultaneously build it from the other opts, but maybe they exist

File: tests/test_cli.py [hunk 4]
```diff
@@ -94,6 +95,58 @@ def test_model_upload(mocker, tmp_path):
...
+    assert '(title: "bar")' in query
+    assert '(description: "foo")' in query
+    assert " AND " in query
+
+
+@pytest.mark.cli
+def test_search_raw_query(mocker):
+    mock_client = mocker.MagicMock(GardenClient)
+    mocker.patch("garden_ai.app.garden.GardenClient").return_value = mock_client
+    mock_rich = mocker.MagicMock(rich)
```
Comment (OwenPriceSkelly) [inline tests/test_cli.py:129]: fully did not know you could mock entire modules. I feel like I've stumbled on a new pytest secret every time for like the past 3-4 PRs

**Review by OwenPriceSkelly — APPROVED** [review]
Looks good to me! 🤠

File: garden_ai/app/garden.py (hunk 3 above)
Comment (WillEngler) [reply garden_ai/app/garden.py:190]: No harm in being super explicit, might as well put it in

**Pull request merged**